### PR TITLE
Update checkbox-provider-base's packaging to 3.3.0~dev19~ubuntu24.04.1

### PR DESCRIPTION
**Pull request by starnight**
Import checkbox-provider-base (3.3.0~dev19~ubuntu24.04.1)'s debian into the debian branch as the commit ("Import debian packaging from checkbox-provider-base 3.3.0~dev19~ubuntu24.04.1") and tagged as `Version_3.3.0_debian`.

1. Squashed the fix automatically.
2. Fix the conflict:
  * debian/rules: `plainbox-provider-checkbox` -> `checkbox-provider-base`
  * debian/control: `python3-checkbox-support (>= 0.25)` -> `python3-checkbox-support`

https://phabricator.endlessm.com/T35403

### CODE REVIEW

File: debian/changelog [hunk 1]
```diff
@@ -1,97 +1,39 @@
-plainbox-provider-checkbox (0.59.0~ppa~ubuntu21.10.1) impish; urgency=low
+checkbox-provider-base (3.3.0~dev19~ubuntu24.04.1) noble; urgency=low
```
Comment (dbnicholson) [inline debian/changelog:1]: You either need to:

1. Keep the old source package name
2. Change the name of the git repo to match the source package name
3. Change the jenkins job so it specifies the [source package name to use](https://github.com/endlessm/eos-build/blob/master/ci-integration/jenkins-jobs/obs.yaml#L260)

File: debian/control [hunk 2]
```diff
@@ -1,28 +1,23 @@
-Source: plainbox-provider-checkbox
+Source: checkbox-provider-base
```
Comment (dbnicholson) [inline debian/control:1]: Similarly here.

**Review by dbnicholson — CHANGES_REQUESTED** [review]

File: debian/changelog (hunk 1 above)
Comment (starnight) [reply debian/changelog:1]: Covered by https://github.com/endlessm/eos-build/pull/1540

**Comment by starnight** [comment]
Jenkins: retest this please

**Pull request merged**